src/main/java/gtPlusPlus/core/util/recipe/RecipeHashStrat.java

```diff
@@ -0,0 +1,110 @@
+package gtPlusPlus.core.util.recipe;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+
+import gnu.trove.strategy.HashingStrategy;
+import gregtech.api.util.GT_Recipe;
+
+public class RecipeHashStrat {
+
+    public static final HashingStrategy<GT_Recipe> RecipeHashingStrategy = new HashingStrategy<>() {
+
+        // Not implemented.
+        @Override
+        public int computeHashCode(GT_Recipe stack) {
+            return 0;
```

Comment:
This means all the objects have the same hashcode, meaning set / map will be basically just a list, making computation time `O(N²)`.
As far as I understand, object hashcode is "allowed" to have collisions. So I made half-baked hashcode implementation: `return Objects.hash(recipe.mDuration, recipe.mEUt);` And this improved performance significantly:
|recipemap|amount of recipes|before|after|
|---|---|---|---|
|centrifuge|1032|1.320 s|197.9 ms|
|mixer|301|186.7 ms|35.45 ms|
|electrolizer|277|81.05 ms|13.65 ms|
|freezer|158|27.11 ms|18.34 ms|